### PR TITLE
Feature/civ 1636

### PR DESCRIPTION
**Pull request by MMNycz**
**Before creating a pull request make sure that:**

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-1636

### Change description ###

Judges Hearing Screen labels need to cater for up to 2 respondents - Put each statement separate: Applicant prefers ABC. Respondent1 prefers CDE. Respondent2 prefers EFG. If a respondent has not replied - skip that statement. Each user will have been assigned a role previously - Applicant, Respondent1 and Respondent2

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by LeonardoPalmeiro — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java [hunk 1]
```diff
@@ -584,16 +611,43 @@ private String getJudgeHearingPrefType(CaseData caseData, YesOrNo isAppAndRespSa
                 .getDisplayedValue());
         }
 
+        if (caseData.getRespondentsResponses() != null && caseData.getRespondentsResponses().size() == 2) {
+            List<Element<GARespondentResponse>> respondentResponce = caseData.getRespondentsResponses();
+            String respondent1Id = caseData.getGeneralAppRespondentSolicitors().get(0).getValue().getId();
+            String respondent2Id = caseData.getGeneralAppRespondentSolicitors().get(1).getValue().getId();
+            if (respondentResponce != null) {
+                Optional<Element<GARespondentResponse>> respondenceElementOptional;
```
Comment (vasudevganesanhmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java:619]: can be respondentElementOptional1 and the other respondentElementOptional2. .
also 620 and 619 can be combined

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java [hunk 2]
```diff
@@ -584,16 +611,43 @@ private String getJudgeHearingPrefType(CaseData caseData, YesOrNo isAppAndRespSa
...
+                respondenceElementOptional = respondentResponce.stream()
+                    .filter(res -> res.getValue() != null && res.getValue().getGaRespondentDetails()
+                        .equals(respondent1Id)).findFirst();
+                Optional<Element<GARespondentResponse>> responseElementOptiona2 = respondentResponce.stream()
+                    .filter(res2 -> res2.getValue() != null && res2.getValue().getGaRespondentDetails()
+                        .equals(respondent2Id)).findFirst();
+                if (respondenceElementOptional.isPresent()) {
+                    respondet1HearingType = respondenceElementOptional.get().getValue()
```
Comment (vasudevganesanhmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java:627]: spelling  .respondent1

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java [hunk 3]
```diff
@@ -631,13 +717,132 @@ private String getJudgeVulnerabilityText(CaseData caseData) {
...
+            String respondent1Id = caseData.getGeneralAppRespondentSolicitors().get(0).getValue().getId();
+            String respondent2Id = caseData.getGeneralAppRespondentSolicitors().get(1).getValue().getId();
+            if (respondentResponce != null) {
+                Optional<Element<GARespondentResponse>> respondenceElemetnOptional1 = respondentResponce.stream()
+                    .filter(res1 -> res1.getValue() != null && res1.getValue().getGaRespondentDetails()
+                        .equals(respondent1Id)).findFirst();
+                Optional<Element<GARespondentResponse>> respondenceElemetnOptional2 = respondentResponce.stream()
+                    .filter(res2 -> res2.getValue() != null && res2.getValue().getGaRespondentDetails()
+                        .equals(respondent2Id)).findFirst();
+                if (respondenceElemetnOptional1.isPresent() && respondenceElemetnOptional2.isPresent()) {
```
Comment (vasudevganesanhmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java:758]: minor spelling

**Comment by vasudevganesanhmcts** [comment]
overall would suggest exploring ways to reduce duplicate code where possible

**Review by vasudevganesanhmcts — APPROVED** [review]

**Pull request merged**